src/effects/effects_base/internal/effectspresetscontroller.cpp

```diff
@@ -107,21 +110,129 @@ Ret EffectsPresetsController::saveCurrentAsPreset(const EffectInstanceId& effect
     return ok ? muse::make_ok() : muse::make_ret(Ret::Code::InternalError);
 }
 
-void EffectsPresetsController::deletePreset(const EffectId& effectId, const PresetId& presetId)
+muse::Ret EffectsPresetsController::deletePreset(const EffectId& effectId, const PresetId& presetId)
 {
-    UNUSED(effectId);
-    UNUSED(presetId);
-    interactive()->warning("Delete Preset", std::string("Not implemented"));
+    IInteractive::Result res = interactive()->question(
+        muse::trc("effects", "Delete Preset"),
+        muse::mtrc("effects", "Are you sure you want to delete \"%1\"?")
+        .arg(au3::wxToString(presetId)).toStdString(),
+        { IInteractive::Button::No, IInteractive::Button::Yes });
+
+    if (res.button() == static_cast<int>(muse::IInteractive::Button::No)) {
+        return muse::make_ret(Ret::Code::Cancel);
+    }
+
+    auto& pluginManager = PluginManager::Get();
+    bool ok = pluginManager.RemoveConfigSubgroup(
+        PluginSettings::Private,
+        au3::wxFromString(effectId),
+        UserPresetsGroup(presetId)
+        );
+
+    if (ok) {
+        m_userPresetsChanged.send(effectId);
+    }
+
+    return ok ? muse::make_ok() : muse::make_ret(Ret::Code::InternalError);
 }
 
-void EffectsPresetsController::importPreset(const EffectId& effectId)
+static std::vector<std::string> presetFilesFilter()
 {
-    UNUSED(effectId);
-    interactive()->warning("Import Preset", std::string("Not implemented"));
+    return { muse::trc("effects", "Presets") + " (*.txt)",
+             muse::trc("global", "All files") + " (*)" };
+}
+
+muse::Ret EffectsPresetsController::importPreset(const EffectInstanceId& effectInstanceId)
+{
+    const EffectId effectId = instancesRegister()->effectIdByInstanceId(effectInstanceId);
+    Effect* effect = effectsProvider()->effect(effectId);
+    IF_ASSERT_FAILED(effect) {
+        return muse::make_ret(Ret::Code::InternalError);
+    }
+
+    EffectSettings* settings = instancesRegister()->settingsById(effectInstanceId);
+    IF_ASSERT_FAILED(settings) {
+        return muse::make_ret(Ret::Code::InternalError);
+    }
+
+    const std::string interactiveTitle = muse::trc("effects", "Import Effect Parameters");
+    io::path_t path = interactive()->selectOpeningFile(QString::fromStdString(interactiveTitle),
+                                                       globalConfiguration()->homePath(),
```

Comment:
A minor question: Is there some machinery that can allow us to save the last directory used for each kind of dialog? 
This is a minor thing that always gets me when the app does not remember what directory I've used before.
Imagine that you have downloaded a pack of presets and then you are importing them one by one and every time you have to navigate to the directory. 

The same goes for audio export and any other file operations. Using the last directory for each kind of open dialog would be really helpful.

Reply:
added last path